### PR TITLE
uplink ttl

### PR DESCRIPTION
**Pull request by cam-a**
Add ability to specify ttl for uploads

https://storjlabs.atlassian.net/browse/V3-1189

### CODE REVIEW

File: cmd/uplink/cmd/cp.go [hunk 1]
```diff
@@ -34,6 +36,7 @@ func init() {
 		RunE:  copyMain,
 	}, RootCmd)
 	progress = cpCmd.Flags().Bool("progress", true, "if true, show progress")
+	ttl = cpCmd.Flags().Duration("ttl", 0, "number of days before file deletion. Default (0) is indefinite")
```
Comment (mniewrzal) [inline cmd/uplink/cmd/cp.go]: Maybe we can name it `expiration` or something like that?

Comment (kaloyan-raev) [reply cmd/uplink/cmd/cp.go]: or `days-to-live`, or `expires-in`.

File: cmd/uplink/cmd/cp.go [hunk 2]
```diff
@@ -34,6 +36,7 @@ func init() {
 		RunE:  copyMain,
 	}, RootCmd)
 	progress = cpCmd.Flags().Bool("progress", true, "if true, show progress")
+	expires = cpCmd.Flags().Int("expires", 0, "number of days before file deletion. Default (0) is indefinite")
```
Comment (egonelbre) [inline cmd/uplink/cmd/cp.go]: This probably should be specified as a `time.Duration` and such that you can specify `1yr`, `1day` etc.

File: cmd/uplink/cmd/cp.go [hunk 3]
```diff
@@ -34,6 +36,7 @@ func init() {
 		RunE:  copyMain,
 	}, RootCmd)
 	progress = cpCmd.Flags().Bool("progress", true, "if true, show progress")
+	expires = cpCmd.Flags().String("expires", "", "optional expiration date of an object")
```
Comment (mniewrzal) [inline cmd/uplink/cmd/cp.go]: can you add example date in a description?

**Comment by cam-a** [comment]
Actually, I think I need to edit this still
From time package doc: 
"... when used with time.Parse they do not accept all the time formats permitted by the RFCs"

**Comment by mniewrzal** [comment]
hmm if it supports major formats then I wouldn't worry at the moment. If main format like `2014-10-01T20:30:00Z` is working then its fine for me. You can merge as is and create jira ticket to improve situation later.

**Pull request merged**